src/utils/issue.ts

```diff
@@ -169,3 +194,38 @@ async function getOpenedPullRequests(context: Context, username: string): Promis
   const prs = await getAllPullRequests(context, "open");
   return prs.filter((pr) => !pr.draft && (pr.user?.login === username || !username));
 }
+
+/**
+ * Extracts the task id from the PR body. The format is:
+ * `Resolves #123`
+ * `Requires https://www.github.com/.../issue/123`
+ * `Fixes https://github.com/.../issues/123`
+ * `Closes #123`
+ * `Depends on #123`
+ * `Related to #123`
+ */
+export function issueLinkedViaPrBody(prBody: string | null, issueNumber: number): boolean {
+  const regex = // eslint-disable-next-line no-useless-escape
+    /(?:Resolves|Fixes|Closes|Depends on|Related to) #(\d+)|https:\/\/(?:www\.)?github.com\/([^\/]+)\/([^\/]+)\/(issue|issues)\/(\d+)|#(\d+)/gi;
+  const matches = prBody?.match(regex);
```

Comment:
The problem I noticed with this, is that in our issue template we have a comment that contains `Resolves #1234`. This matches the regex as well because comments are not filtered out, which I believe is also the case here.

Reply:
hmm, the `"pull_request" in source` check should eliminate any issues from the timeline afaik

But both of these failed to hit my timeline when I was QA-ing and I'm clueless as to why other than they are in a different repo. Unsure how to proof things better without plugging it into V2 for testing but that's not ideal

https://github.com/ubq-testing/bot-ai/pull/48
https://github.com/ubq-testing/bot-ai/issues/47

Reply:
I meant that the `prBody` would be like:
```
Resolves #12

<!-- Resolves #1234 -->
```

Second one is a comment and should be ignored.

Reply:
This issue also exists in other repos sadly.

Reply:
what do you suggest then, avoid using this sort of approach all together and revert back to my first approach or solidify it?

Reply:
Solved for html comments but even still I expect to see some teething problems with this especially with any of the selectors other than "Resolves"

Reply:
I think a quick fix that would be reliable is to remove comment, by using a substring with a regex, so comments are ignored, then your logic should be good.

Reply:
I always open up my pull body with "Resolves #0". Are we sure that GitHub correctly links the pull if "Resolves" is located elsewhere? 

`^Resolves\s#\d+?\D`

Reply:
It does resolve it properly wherever it is located, except HTML comments that are also properly ignored.